stock_picking_batch_planner/hooks.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2025 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo.tools.sql import column_exists, create_column
+
+
+def pre_init_hook(env):
+    # Create the column `auto_batch_original` if it does not exist and copy the value of
+    # `auto_batch` to it for all existing picking types.
+    if not column_exists(env.cr, "stock_picking_type", "auto_batch_original"):
+        create_column(env.cr, "stock_picking_type", "auto_batch_original", "boolean")
+    env.cr.execute("""UPDATE stock_picking_type SET auto_batch_original = auto_batch""")
+
+    # Ensure that the value of `plan_batch` is correct according to the value of
+    # `auto_batch_original` for all existing picking types.
+    if not column_exists(env.cr, "stock_picking_type", "plan_batch"):
+        create_column(env.cr, "stock_picking_type", "plan_batch", "boolean")
+    env.cr.execute("""UPDATE stock_picking_type SET plan_batch = auto_batch_original""")
```

Comment:
suggestion: batch both updates:


```suggestion
    # Create the missing columns `auto_batch_original` and `plan_batch`
    if not column_exists(env.cr, "stock_picking_type", "auto_batch_original"):
        create_column(env.cr, "stock_picking_type", "auto_batch_original", "boolean")
    if not column_exists(env.cr, "stock_picking_type", "plan_batch"):
        create_column(env.cr, "stock_picking_type", "plan_batch", "boolean")
    # Fill their values
    env.cr.execute(
      """
      UPDATE stock_picking_type
      SET auto_batch_original = auto_batch, plan_batch = auto_batch
      """
    )
```

Reply:
Done